src/llm_tracekit/bedrock/invoke_agent.py

```diff
@@ -191,46 +192,49 @@ def _process_usage_data(self, usage: Dict[str, int]):
     def _process_chat_history(self, raw_messages: List[Dict[str, Any]]):
         try:
             prompt_history: List[Message] = []
-            completion_history: List[Choice] = []
-            local_tool_calls_buffer: List[ToolCall] = []
-
+            last_tool_call_id: str = None
             for msg in raw_messages:
                 role = msg.get("role")
                 raw_content = msg.get("content", "")
 
-                if role is None or "type=tool_result" in raw_content:
+                if role is None:
                     continue
+                
+                content = parsing_utils.parse_content(raw_content)
 
-                tool_call = parsing_utils.parse_tool_use(raw_content)
-                if tool_call is not None:
-                    local_tool_calls_buffer.append(tool_call)
+                if "type=tool_use" in raw_content:
+                    tool_call = parsing_utils.parse_tool_use(raw_content)
+                    if tool_call is not None:
+                        last_tool_call_id = tool_call.id
```

Comment:
* storing this is redundant (and potentially buggy), you can extract the tool_call_id from the tool_result data
* what happens if the agent requests more than 1 tool_use in the same message?